File: py-basics/src/lectures/classes/smart_traffic_lights.py
# 6kyu | Smart Traffic Lights
# https://www.codewars.com/kata/58587905ed1b4dad6e0000c6/python

class SmartTrafficLight:
    def __init__(self, st1, st2):
        pass

    def turngreen(self):
        return


def test(actual, expected):
    checkmark = '✔️' if expected == actual else '❌'
    print(f"{checkmark} expected -> actual: {str(expected)} -> {str(actual)}")


t = SmartTrafficLight([42, '27th ave'], [72, '3rd st'])
test(t.turngreen(), '3rd st')
test(t.turngreen(), '27th ave')
test(t.turngreen(), None)

File: js-basics/src/lectures/classes/smart-traffic-lights.js
/**
 * 6kyu | Smart Traffic Lights
 * https://www.codewars.com/kata/58587905ed1b4dad6e0000c6/javascript
 */

class SmartTrafficLight {
    constructor(st1, st2) {
        // TODO
    }
    turngreen() {
        return null;
    }
}

function test(actual, expected) {
    let checkmark = expected == actual ? '✔️' : '❌';
    console.log(`${checkmark} expected -> actual: ${new String(expected)} -> ${new String(actual)}`);
}

let t = new SmartTrafficLight([42, '27th ave'], [72, '3rd st']);
test(t.turngreen(), '3rd st');
test(t.turngreen(), '27th ave');
test(t.turngreen(), null);
